docs/usage.md

```diff
@@ -101,14 +101,14 @@ In this case, there is no `match` key, so only one rule needs to match:
 `rules` is an array of rules that can be used to target specific files with your defined syntax file.  The rules are processed until the first rule matches, so order your rules in a way that makes sense to you.
 
 ### Filename Rule
-Filename rule defines a filename with regex.  This uses the full file path.
+Filename rule defines a filename with regex.  The pattern is matched against the beginning of the full file path (there is an implicit `^`).
```

Comment:
> It actually does make a difference whether you use $ in these rules; e.g., ".*\\.php" will match /a/b/c/foo.php5, where ".*\\.php$". 

You are totally right.  Even in our rules we don't always use `$` to terminate matches (especially for first_line rules).  I have no idea why I was thinking otherwise as I heavily use that functionality in other projects.  Never mind, carry on :smile:.

> Thanks, i was afraid you'd think i was nit-picking. I have just recently converted from MacVim, so maybe as i get more into the Sublime ecosystem i can work on this kind of thing more.

Nah, I maintain a lot of Sublime plugins, I honestly wish more people would attempt pulls; every little bit helps.